airbyte-webapp/src/packages/cloud/services/users/InviteUsersModalService.tsx

```diff
@@ -19,7 +19,9 @@ export const useInviteUsersModalService = () => {
   return ctx;
 };
 
-export const InviteUsersModalServiceProvider: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
+export const InviteUsersModalServiceProvider: React.FC<
+  React.PropsWithChildren<unknown> & { invitedFrom: "source" | "destination" | "user.settings" }
```

Comment:
We try to always exclude properties into their own interface, i.e. as a `InviteUsersModalServiceProviderProps`. Also that interface, should be used as the generic type for `PropsWithChildren` instead of intersected with it, i.e. `React.FC<React.PropsWithChildren<InviteUsersModalServiceProviderProps>>`

Reply:
I'll make the change. 
Could you explain why we prefer to/should use the interface as the generic type? 
I'm super interested in learning more about Typescript as I'm fairly new to it and don't usually use more than basics.

Reply:
The `React.PropsWithChildren`'s sole purpose is basically to wrap the properties and add `children` to those props, so doing `React.PropsWithChildren<unknown> & { ... }` is kind of like saying: I have the perfect template for a shopping list at home, and when going to the shop I take that (empty) template and my actual list ("Manolitos, Manolitos, Manolitos") that I wrote on a completely blank second paper. (I'll let you know once I come up with a better analogy.)

Reply:
Great analogy! Totally understood! thanks for taking the time to explain it 😊